pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -172,118 +174,123 @@ type CertificateRequestPolicyAllowedX509Subject struct {
 	// +optional
 	PostalCodes *CertificateRequestPolicyAllowedStringSlice `json:"postalCodes,omitempty"`
 
-	// SerialNumber defines the X.509 Subject Serial Number that may be requested
-	// for.
+	// SerialNumber defines the X.509 Subject Serial Number that may be
+	// requested for.
 	// +optional
 	SerialNumber *CertificateRequestPolicyAllowedString `json:"serialNumber,omitempty"`
 }
 
-// CertificateRequestPolicyAllowedStringSlice represents an allowed string
-// slice value paired with whether the field is a required value on the
-// request.
+// CertificateRequestPolicyAllowedStringSlice represents allowed string values
+// paired with whether the field is a required value on the request.
+// An empty Values paired with Required `true` is an "always-false" condition.
+// always denies.
+// The validating webhook enforces that Values cannot be omitted if Required
+// is `true`.
 type CertificateRequestPolicyAllowedStringSlice struct {
-	// Defines the values that are permissible to be present on request.
+	// Values defines allowed attribute values on the related
+	// CertificateRequest field.
 	// Accepts wildcards "*".
-	// An omitted field or value of `nil` forbids any value on the related field
-	// in the request from being requested.
-	// An empty slice `[]` is equivalent to `nil`, however an empty slice paired
-	// with Required `true` is an impossible condition that always denies.
-	// Values may not be `nil` if Required is `true`.
+	// If set, the related field must contain a subset of the allowed values.
```

Comment:
Must it be a proper subset? Or does this just mean it can't be a superset / have items not in this list?

Reply:
> A subset is a set that has either some or all of the elements of another set, called the superset.

I am not sure what is the correct wording here. 🤔 Maybe "can only have values matching at least one item in the list"?